frappe/hooks.py

```diff
@@ -166,6 +164,8 @@
 		"on_change": [
 			"frappe.social.doctype.energy_point_rule.energy_point_rule.process_energy_points",
 			"frappe.automation.doctype.milestone_tracker.milestone_tracker.evaluate_milestone",
+			"frappe.automation.doctype.assignment_rule.assignment_rule.apply",
+			"frappe.automation.doctype.assignment_rule.assignment_rule.update_due_date",
```

Comment:
Note: This will trigger on `db_set` too.

Reply:
I guess that isn't bad, at least for the assignment rule. 🤔

Reply:
Changed to run `on_cancel` and `on_update_after_submit` only